src/network/connection.js

```diff
@@ -199,6 +199,7 @@ module.exports = class Connection {
     }
 
     this.logDebug('disconnecting...')
+    this.requestQueue.destroy();
```

Comment:
@Nevon  Just to make sure that it works correctly. Do we expect that new Connection will be always created on reconnect ? Otherwise we can get situation when we cleared interval and we still reuse the same connection.

Reply:
It should be the case. Maybe in the setInterval we could check `this.connected` of the connection, and clear the interval if we are no longer connected?

Reply:
Good idea i think this will make sure to reset interval properly, will also timeout all inflight requests if we are not connected.

Reply:
I have tried adding isConnected handler, however not sure if it is worth of effort as before system is connected interval is running which will cancel immediately, we may need to move RequestQueue instantiation in to on connect handler (or have another function to trigger the interval in the RequestQueue). 

I can implement above logic if you would like to :) 

I have skimmed the code and looks like overall we correctly trigger disconnect which should clean up this interval.

Reply:
What makes me a little uncomfortable is that the `RequestQueue` is instantiated in the constructor of the connection, not in the `connect`, so as soon as the object is created the setInterval is started. I don't think that we call `disconnect` in every single of those cases (unit tests, for example). And even if we do, if we aren't connected, `disconnect` returns early, so the `setInterval` would not be cleared.

I think just passing in a function to the `RequestQueue` from the connection to check if the interval should be cleared is simple enough:

```js
this.requestQueue = new RequestQueue({
  getConnection: () => this
})
```

```js
class RequestQueue {
  constructor({ getConnection }) {
    if (this.enforceRequestTimeout) {
        const requestTimeoutCheckInterval = Math.min(this.requestTimeout, 100)
        this.requestTimeoutIntervalId = setInterval(() => {
            this.inflight.forEach(request => {
                if (Date.now() - request.sentAt > request.requestTimeout) {
                    request.timeoutRequest()
                }
            })
            const connection = getConnection()
            if (!connection || !connection.isConnected()) {
                clearInterval(this.requestTimeoutIntervalId)
            }
        }, requestTimeoutCheckInterval)
    }
  }
}
```

Reply:
@Nevon in this case we will clear the interval even before the system is actually connected as by default `this.connected` is `false` and if we are unable to connect (call `onConnect` in `Connection` class) within the first interval call it will clear the interval. May be it is better to move RequestQueue creation in to `onConnect` ? Or move `setInterval` logic in to separate function and call it from `onConnect` ? 

like:

```js
  const onConnect = () => {
        clearTimeout(timeoutId)
        this.connected = true
        this.requestQueue.scheduleRequestTimeoutCheck()
        resolve(true)
      }
```

Reply:
Right, of course 🤦‍♂️

Try moving it into connect. I'm wondering if there's a case for adding things to the queue before connecting, but I'm not sure.

Reply:
@Nevon had a look at `send` method looks like we have `failIfNotConnected` so there should not be any way to push in the queue before connection. I would try to move queue creation in to onConnect

Reply:
@Nevon i have done it slightly different way it will also work properly and in my opinion suits better. Would like you to check it out and let me know if anything should be changed.